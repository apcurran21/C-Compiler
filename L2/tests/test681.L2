(@main
  (@main
    0
    :call_label0
    %radius <- 5
    rdi <- %radius

mem rsp -8 <- :call_label1
    call @circ 1

:call_label1
    %value <- rax
    %value <- %value
    %value <<= 1
    %value <- %value
    %value += 1
    rdi <- %value
    call print 1
    rdi <- %radius

mem rsp -8 <- :call_label2
    call @area 1

:call_label2
    %value <- rax
    %value <- %value
    %value <<= 1
    %value <- %value
    %value += 1
    rdi <- %value
    call print 1
    rdi <- %radius

mem rsp -8 <- :call_label3
    call @SA 1

:call_label3
    %value <- rax
    %value <- %value
    %value <<= 1
    %value <- %value
    %value += 1
    rdi <- %value
    call print 1
    rdi <- %radius

mem rsp -8 <- :call_label4
    call @vol 1

:call_label4
    %value <- rax
    %value <- %value
    %value <<= 1
    %value <- %value
    %value += 1
    rdi <- %value
    call print 1
    return
  )
  (@circ
    1
    %r <- rdi
    :call_label_ciao_5
    %pi <- 3
    %d <- 2
    %d *= %r
    %c <- %d
    %c *= %pi
    rax <- %c
    return
  )
  (@area
    1
    %r <- rdi
    :call_label_ciao__ciao_6
    %pi <- 3
    %r2 <- %r
    %r2 *= %r
    %a <- %r2
    %a *= %pi
    rax <- %a
    return
  )
  (@SA
    1
    %r <- rdi
    :call_label_ciao__ciao__ciao_7
    %pi <- 3
    %r2 <- %r
    %r2 *= %r
    %r2 <- %r2
    %r2 *= 4
    %sa <- %r2
    %sa *= %pi
    rax <- %sa
    return
  )
  (@vol
    1
    %r <- rdi
    :call_label_ciao__ciao__ciao__ciao_8
    %factor <- 4
    %r2 <- %r
    %r2 *= %r
    %r3 <- %r2
    %r3 *= %r
    %v <- %r3
    %v *= %factor
    rax <- %v
    return
  )
)
