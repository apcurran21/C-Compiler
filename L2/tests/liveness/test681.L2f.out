(
(in
(r12 r13 r14 r15 rbp rbx rdi rsi)
(%numer r12 r13 r14 r15 rbp rbx rsi)
(%denom %numer r12 r13 r14 r15 rbp rbx)
(%denom %numer r12 r13 r14 r15 rbp rbx)
(%denom %n r12 r13 r14 r15 rbp rbx)
(%d %n r12 r13 r14 r15 rbp rbx)
(%d %enc_temp %n r12 r13 r14 r15 rbp rbx)
(%d %n r12 r13 r14 r15 rbp rbx %temp)
(%d %n r12 r13 r14 r15 rbp rbx %temp)
(%d %enc_quotient %n r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_1 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_2 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%quotient r12 r13 r14 r15 rbp rbx)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_3 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%quotient r12 r13 r14 r15 rbp rbx)
(%quotient r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%numer r12 r13 r14 r15 rbp rbx rsi)
(%denom %numer r12 r13 r14 r15 rbp rbx)
(%denom %numer r12 r13 r14 r15 rbp rbx)
(%denom %n r12 r13 r14 r15 rbp rbx)
(%d %n r12 r13 r14 r15 rbp rbx)
(%d %enc_temp %n r12 r13 r14 r15 rbp rbx)
(%d %n r12 r13 r14 r15 rbp rbx %temp)
(%d %n r12 r13 r14 r15 rbp rbx %temp)
(%d %enc_quotient %n r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_1 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_2 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%quotient r12 r13 r14 r15 rbp rbx)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%cond_3 %d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%d %n %quotient r12 r13 r14 r15 rbp rbx %temp)
(%quotient r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
